### PR TITLE
realm domains: improve doc text

### PR DESCRIPTION
**Pull request by pvoborni**
It is quite unclear how realm domains behave without reading source
code. New doc text describes its purpose and how it is managed.

### CODE REVIEW

**Comment by pvoborni** [comment]
@abbra  I was also looking at documentation: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html-single/windows_integration_guide/#trust-realmdomain and the first paragraph there seems quite misleading to me. Do you think the wording there is good?

**Comment by abbra** [comment]
You are right that the first paragraph in the docs is not good. It should be removed completely, as well as the second one. I think a text you added in this PR should be used instead.

**Comment by abbra** [comment]
LGTM

**Comment by pvoborni** [comment]
Added a ticket to commit message so that it can be backported to 4.6.  https://pagure.io/freeipa/issue/7424

File: ipaserver/plugins/realmdomains.py [hunk 1]
```diff
@@ -118,10 +128,22 @@ class realmdomains(LDAPObject):
...
 class realmdomains_mod(LDAPUpdate):
-    __doc__ = _('Modify realm domains.')
+    __doc__ = _("""
+    Modify realm domains
+
+    DNS check: When adding domain to the list manually, a DNS check is
+    performed by default. It checks whether the domain is associated
+    with the IPA realm. In practice it checks whether the domain has
+    _kerberos TXT record containing IPA realm name. This check can be
+    skipped by specifying --force option.
```
Comment (flo-renaud) [inline ipaserver/plugins/realmdomains.py]: nitpicking, but I would rephrase:
When manually adding a domain to the list, a DNS check is performed by default. It ensures that the domain is associated with the IPA realm, by checking whether the domain has a _kerberos TXT record containing the IPA realm name. This check can be skipped by specifying --force option.

File: ipaserver/plugins/realmdomains.py [hunk 2]
```diff
@@ -36,6 +36,16 @@
 
 Manage the list of domains associated with IPA realm.
 
+This list is useful for Domain Controllers from other realms which have
+established trust with this IPA realm. They need the information to know
+which request should be forwarded to KDC of this IPA realm.
+
+Automatic management: a domain is added to realm domains list automatically
+when new DNS Zone managed by IPA is created. Same applies for deletion.
```
Comment (flo-renaud) [inline ipaserver/plugins/realmdomains.py]: nitpicking, but I would rephrase:
a domain is automatically added to the realm domains list when a new DNS Zone managed by IPA is created. Same applies for deletion.

File: ipaserver/plugins/realmdomains.py [hunk 3]
```diff
@@ -36,6 +36,16 @@
...
+
+Externally managed DNS: domains which are not managed in IPA server DNS needs
+to be added to the list manually using `ipa realmdomains-mod` command.
```
Comment (flo-renaud) [inline ipaserver/plugins/realmdomains.py]: nitpicking, but I would rephrase:
domains which are not managed in IPA server DNS need to be manually added to the list using `ipa realmdomains-mod` command.

File: ipaserver/plugins/realmdomains.py (hunk 1 above)
Comment (abbra) [reply ipaserver/plugins/realmdomains.py]: This got me to another point: Active Directory domains never have _kerberos TXT record associated, so the check is inadequate for that use case. We should use a different method by looking up for MSDCS entries in the domain and its parents. However, I'm not sure where we should stop. See https://technet.microsoft.com/en-us/library/cc759550(WS.10).aspx for details

**Comment by pvoborni** [comment]
@flo-renaud thanks for the review, PR updated, all nitpicks should be addressed.

**Comment by flo-renaud** [comment]
@pvoborni 
thanks for the update. The doc strings look good now.
As a follow-up, I will open a doc bug in order to fix Windows Integration guide and a pagure ticket linked to @abbra comment about not checking TXT records for AD.

**Comment by pvoborni** [comment]
@flo-renaud , thanks for opening the bug

**Comment by Tiboris** [comment]
master:

* b43e73143de66b0068cd01379c134d5ef6e304e9 realm domains: improve doc text
